packages/p2p/src/components/modal-manager/__tests__/modal-manager-context-provider.spec.tsx

```diff
@@ -0,0 +1,362 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ModalManagerContextProvider from '../modal-manager-context-provider';
+import ModalManager from '../modal-manager';
+import { isDesktop } from '@deriv/shared';
+import { useStores } from 'Stores/index';
+import { MockBuySellModal, MockMyAdsDeleteModal, MockEditAdCancelModal, MockPage } from '../__mocks__/mock-modal-manager-context-provider';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isDesktop: jest.fn(() => true),
+}));
+
+const mock_store: DeepPartial<ReturnType<typeof useStores>> = {
+    general_store: {
+        saveFormState: jest.fn(),
+        setSavedFormState: jest.fn(),
+        setFormikRef: jest.fn(),
+    },
+};
+
+jest.mock('Stores', () => ({
+    ...jest.requireActual('Stores'),
+    useStores: jest.fn(() => mock_store),
+}));
+
+jest.mock('Constants/modals', () => ({
+    Modals: {
+        BuySellModal: (props: any) => <MockBuySellModal {...props} />,
+        MyAdsDeleteModal: (props: any) => <MockMyAdsDeleteModal {...props} />,
+        EditAdCancelModal: (props: any) => <MockEditAdCancelModal {...props} />,
+    },
+}));
+
+describe('<ModalManagerContextProvider />', () => {
+    beforeEach(() => {
+        jest.resetModules();
+    });
+    afterAll(() => {
+        jest.resetModules();
+        jest.resetAllMocks();
+    });
+
+    it('should render the mock modal when showModal is called in desktop view', () => {
+        render(
+            <React.Fragment>
+                <div id='modal_root' />
+                <ModalManagerContextProvider>
+                    <ModalManager />
+                    <MockPage />
+                </ModalManagerContextProvider>
+            </React.Fragment>
+        );
+
+        const showBtn = screen.getByRole('button', {
+            name: /Show BuySellModal/,
+        });
+
+        userEvent.click(showBtn);
+
+        const text = screen.getByText('BuySellModal');
+        expect(text).toBeInTheDocument();
+    });
+
+    it('should render the latest shown modal when showModal is called multiple times in desktop view', () => {
+        render(
+            <React.Fragment>
+                <div id='modal_root' />
+                <ModalManagerContextProvider>
+                    <ModalManager />
+                    <MockPage />
+                </ModalManagerContextProvider>
+            </React.Fragment>
+        );
+
+        const showBuySellModalBtn = screen.getByRole('button', {
```

Comment:
Please follow the naming convention for variables. 
```suggestion
        const buy_sell_modal_btn = screen.getByRole('button', {
```